pkg/tnf/operator/starter.go

```diff
@@ -56,24 +58,43 @@ func HandleDualReplicaClusters(
 	runTnfResourceController(ctx, controllerContext, kubeClient, dynamicClient, operatorClient, kubeInformersForNamespaces)
 
 	// we need node names for assigning auth jobs to specific nodes
+	var once sync.Once
 	klog.Infof("watching for nodes...")
 	_, err := controlPlaneNodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
-			node, ok := obj.(*corev1.Node)
+			_, ok := obj.(*corev1.Node)
 			if !ok {
 				klog.Warningf("failed to convert node to Node %+v", obj)
 			}
-			runTnfAuthJobController(ctx, node.GetName(), controllerContext, operatorClient, kubeClient, kubeInformersForNamespaces)
-			runTnfAfterSetupJobController(ctx, node.GetName(), controllerContext, operatorClient, kubeClient, kubeInformersForNamespaces)
+			// ensure we have both control plane nodes before creating jobs
+			nodeList, err := kubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{
```

Comment:
I know it's premature optimization, but you can also create a lister on top of the informer indexer:

> import corev1listers "k8s.io/client-go/listers/core/v1"
> controlPlaneNodeLister := corev1listers.NewNodeLister(controlPlaneNodeInformer.GetIndexer())


so that way you can also save yourself the additional synchronous list call.